res/controllers/Numark-Scratch-scripts.js

```diff
@@ -0,0 +1,417 @@
+            }
+        });
+
+        // Disable all buttons in the other unit
+        NumarkScratch.effect[1 - (deckNumber - 1)].effectButtons.forEach((effectButton) => {
+            engine.setValue(effectButton.group, effectButton.inKey, false);
+        });
+    };
+
+    this.effectButtons = [];
+    for (let i = 0; i < 3; i++) {
+        this.effectButtons[i] = new components.Button({
+            group: `[EffectRack1_EffectUnit${deckNumber}_Effect${i + 1}]`,
+            midi: [0x98 + deckNumber - 1, (deckNumber - 1) * 3 + i],
+            inKey: "enabled",
+            outKey: "enabled", // Bind the LED state to the effect enabled state
+            shift: function() {
+                this.input = components.Button.prototype.input;
+                this.type = components.Button.prototype.types.toggle;
+            },
+            unshift: function() {
+                this.input = inputUnshifted;
+            },
+        });
+    }
+
+    this.paddle = new components.Button({
+        group: `[EffectRack1_EffectUnit${deckNumber}]`,
+        outKey: "enabled",
+        inKey: `group_[Channel${deckNumber}]_enable`,
+        input: function(channel, control, value, _status, _group) {
+            this.isPress = value !== 0;
+            engine.setValue("[EffectRack1_EffectUnit1]", `group_[Channel${deckNumber}]_enable`, this.isPress);
+            engine.setValue("[EffectRack1_EffectUnit2]", `group_[Channel${deckNumber}]_enable`, this.isPress);
+            this.trigger();
+        }
+    });
+
+    this.dryWetKnob = new components.Pot({
+        group: `[EffectRack1_EffectUnit${deckNumber}]`,
+        inKey: "mix",
+        shift: function() {
+            this.inKey = "super1";
+        },
+        unshift: function() {
+            this.inKey = "mix";
+        },
+    });
+};
+NumarkScratch.EffectUnit.prototype = new components.ComponentContainer();
+
+NumarkScratch.XfaderContainer = function() {
+    this.crossfader = new components.Pot({
+        midi: [0xBF, 0x08],
+        group: "[Master]",
+        inKey: "crossfader",
+    });
+
+    this.setCurve = function(channel, control, value, _status, _group) {
+        switch (value) {
+        case 0x00: // Additive/Linear
+            engine.setValue("[Mixer Profile]", "xFaderMode", 0);
+            engine.setValue("[Mixer Profile]", "xFaderCalibration", 0.4);
+            engine.setValue("[Mixer Profile]", "xFaderCurve", 0.9);
+            break;
+        case 0x7F:  // Picnic Bench/Fast Cut
+            engine.setValue("[Mixer Profile]", "xFaderMode", 0);
+            engine.setValue("[Mixer Profile]", "xFaderCalibration", 0.9);
+            engine.setValue("[Mixer Profile]", "xFaderCurve", 7.0);
+            break;
+        }
+    };
+
+    this.xFaderReverse = function(channel, control, value, _status, _group) {
+        // 0x7F is ON, 0x00 is OFF
+        engine.setValue("[Mixer Profile]", "xFaderReverse", (value === 0x7F) ? 1 : 0);
+    };
```

Comment:
```suggestion
    this.xFaderReverse = new components.Button({
        group: "[Mixer Profile]",
        inKey: "xFaderReverse"
    });
```
and don't forget to adjust the `<key>` in the XML.

Reply:
looks cleaner. done